src/schemes/fluid/weakly_compressible_sph/state_equations.jl

```diff
@@ -49,3 +49,50 @@ function inverse_state_equation(state_equation::StateEquationCole, pressure)
 
     return reference_density * tmp^(1 / exponent)
 end
+
+@doc raw"""
+    StateEquationIdealGas(;sound_speed, reference_density, gamma, background_pressure=0.0,
+                           clip_negative_pressure=false)
+
+Equation of state to describe the relationship between pressure and density
+of a gas using the Ideal Gas Law.
+
+# Keywords
+- `sound_speed`            : Artificial speed of sound.
+- `reference_density`      : Reference density of the fluid.
+- `gamma`                  : Heat-capacity ratio
+- `background_pressure=0.0`: Background pressure.
+"""
+struct StateEquationIdealGas{ELTYPE, CLIP}
+    sound_speed         :: ELTYPE
+    reference_density   :: ELTYPE
+    gamma               :: ELTYPE
+    background_pressure :: ELTYPE
+
+    function StateEquationIdealGas(; sound_speed, reference_density, gamma,
+                                   background_pressure=0.0, clip_negative_pressure=false)
+        new{typeof(sound_speed), clip_negative_pressure}(sound_speed, reference_density,
+                                                         gamma, background_pressure)
+    end
+end
+
+clip_negative_pressure(::StateEquationIdealGas{<:Any, CLIP}) where {CLIP} = CLIP
+
+function (state_equation::StateEquationIdealGas)(density)
+    (; reference_density, sound_speed, gamma, background_pressure) = state_equation
+    pressure = (density - reference_density) * sound_speed^2 / gamma + background_pressure
+
+    # This is determined statically and has therefore no overhead
+    if clip_negative_pressure(state_equation)
+        return max(0.0, pressure)
+    end
+
+    return pressure
+end
```

Comment:
This function is almost identical with `StateEquationCole` and `exponent=1` (described here in the [docs](https://trixi-framework.github.io/TrixiParticles.jl/dev/systems/weakly_compressible_sph/#equation_of_state)) except the `gamma` in the bulk modulus.

Or Am I completely missing something?

I don't want to hold back the review process, but why not adding a kwarg `gamma` to the `StateEquationCole` instead of creating a new object?. 
The default is then `gamma=exponent` but can be overwritten.

Reply:
While this is true I am not sure this makes a lot of sense. 
1. Everyone expects there to be an ideal gas state equation when you calculate a gas.
2. This is really quirky because it would require some error checks. E.g. if gamma and exponent are set etc...
3. As soon as there is heat flux the equation changes to versions with temperature (which were removed because they are unused at this time)

Reply:
Agreed. From a user's perspective, this is more intuitive.

Reply:
Alright, thanks for the explanation.